File: spring-boot/src/main/java/com/example/demo/service/backend/AdminService.java
package com.example.demo.service.backend;

import com.example.demo.dto.request.backend.CreateUserRequest;
import com.example.demo.dto.response.UserResponse;
import com.example.demo.enums.UserRole;

import java.util.List;
import java.util.Map;

/**
 * 管理員服務接口
 * 處理後台管理相關業務邏輯
 */
public interface AdminService {

    /**
     * 創建管理員帳號
     */
    UserResponse createAdmin(CreateUserRequest request, String createdBy);

    /**
     * 創建前台用戶帳號
     */
    UserResponse createUser(CreateUserRequest request, String createdBy);

    /**
     * 獲取所有用戶
     */
    List<UserResponse> getAllUsers();

    /**
     * 根據角色獲取用戶
     */
    List<UserResponse> getUsersByRole(UserRole role);

    /**
     * 切換用戶啟用狀態
     */
    UserResponse toggleUserStatus(Long userId, String operatorEmail);

    /**
     * 軟刪除用戶
     */
    void deleteUser(Long userId, String deletedBy);

    /**
     * 恢復已刪除的用戶
     */
    UserResponse restoreUser(Long userId, String restoredBy);

    /**
     * 獲取用戶統計信息
     */
    Map<String, Object> getUserStatistics();

    /**
     * 獲取已刪除的用戶列表
     */
    List<UserResponse> getDeletedUsers();

    /**
     * 為用戶添加角色
     */
    UserResponse addUserRole(Long userId, UserRole role, String operatorEmail);

    /**
     * 移除用戶角色
     */
    UserResponse removeUserRole(Long userId, UserRole role, String operatorEmail);
}

File: react/src/services/backend/adminService.js
import api from '../api';

export const adminService = {
    // 管理員登入
    login: async (credentials) => {
        return await api.post('/admin/v1/auth/login', credentials);
    },

    // 創建管理員
    createAdmin: async (userData) => {
        return await api.post('/admin/v1/create-admin', userData);
    },

    // 創建用戶
    createUser: async (userData) => {
        return await api.post('/admin/v1/create-user', userData);
    },

    // 獲取所有用戶
    getAllUsers: async () => {
        return await api.get('/admin/v1/users');
    },

    // 根據角色獲取用戶
    getUsersByRole: async (role) => {
        return await api.get(`/admin/v1/users/role/${role}`);
    },

    // 獲取所有管理員
    getAllAdmins: async () => {
        return await api.get('/admin/v1/admins');
    },

    // 獲取所有前台用戶
    getAllFrontendUsers: async () => {
        return await api.get('/admin/v1/frontend-users');
    },

    // 切換用戶狀態
    toggleUserStatus: async (userId) => {
        return await api.put(`/admin/v1/users/${userId}/toggle-status`);
    },

    // 軟刪除用戶
    deleteUser: async (userId) => {
        return await api.delete(`/admin/v1/users/${userId}`);
    },

    // 恢復已刪除的用戶
    restoreUser: async (userId) => {
        return await api.put(`/admin/v1/users/${userId}/restore`);
    },

    // 為用戶添加角色
    addUserRole: async (userId, role) => {
        return await api.put(`/admin/v1/users/${userId}/add-role/${role}`);
    },

    // 移除用戶角色
    removeUserRole: async (userId, role) => {
        return await api.put(`/admin/v1/users/${userId}/remove-role/${role}`);
    },

    // 獲取已刪除的用戶
    getDeletedUsers: async () => {
        return await api.get('/admin/v1/deleted-users');
    },

    // 獲取用戶統計信息
    getUserStatistics: async () => {
        return await api.get('/admin/v1/statistics');
    }
};